Add UNIX pipe concept to linux-exercises.md

vkuznet opened the pull request:
Add concept of the pipe and corresponding excercises

vkuznet commented:
@wernersun please review and merge

wernersun commented:
Thanks @vkuznet -- a few comments:
- For definition of echo/cat/grep/env/more/less, add reference to https://xcitecourse.org/theme2/SF100/
- Is it necessary to pipe `echo` into `cat`? Seems like you get the same output without `cat`?
- Instead of `/tmp`, perhaps use CHESS user directory from Exercise 3?
- What is the meaning of the blue diamond in front of Example 2?

vkuznet commented:
@wernersun , I addressed your feedback. In particular:
- renamed UNIX to Linux
- expanded `echo` and `cat` examples, of course no need for cat after `echo` but it does not change anything either, just illustration of the pipe
- I explained what `/tmp` area mean and I still prefer to use it as a general concept
- The diamond was oversight from my side using markdown editor

Please review again, and if necessary merge it.

wernersun commented:
Thanks @vkuznet. For `echo|cat`, I worry that it will be confusing for beginners -- if they first type `echo`, then `echo|cat`, they will see no difference in the output, so what is the point of the pipe? They may also blindly use `echo|cat` in the future, thinking that is the way they are supposed to do it. I think the examples should be meaningful.

vkuznet commented:
@wernersun , I already addressed this and replaced `echo | cat` with `echo | grep`, see pipes exercise 1 in this PR. And, I also provided examples of using cat for viewing and creating files.

wernersun commented:
@vkuznet, yes I like Exercise 1 and it works well. I'm talking about "Example 1: Pass text from echo to cat" ` echo "Hello from Linux pipe!" | cat`

Maybe we don't need the 2 "examples", just move directly into the 4 "exercises"?
